src/dfx/src/commands/canister/create.rs

```diff
@@ -102,14 +115,25 @@ pub async fn exec(
         && !matches!(call_sender, CallSender::Wallet(_))
         && !network.is_playground()
     {
-        let wallet = get_or_create_wallet_canister(
+        match get_or_create_wallet_canister(
             env,
             env.get_network_descriptor(),
             env.get_selected_identity().expect("No selected identity"),
         )
-        .await?;
-        proxy_sender = CallSender::Wallet(*wallet.canister_id_());
-        call_sender = &proxy_sender;
+        .await
+        {
+            Ok(wallet) => {
+                proxy_sender = CallSender::Wallet(*wallet.canister_id_());
+                call_sender = &proxy_sender;
+            }
+            Err(err) => {
+                if CYCLES_LEDGER_ENABLED {
+                    debug!(env.get_logger(), "No wallet configured.");
```

Comment:
should this look for `GetOrCreateWalletCanisterError::NoWalletConfigured`, rather than hiding any other error?